block_course_contents.php

```diff
@@ -164,15 +164,41 @@ public function get_content() {
                 $title = $format->get_section_name($section);
             }
 
+            if ( ($i == 0) && (!empty($globalconfig->display_course_link)) ) {
+                $sectionclass = 'section-item';
+
+                if ((!isset($selected)) && (empty($selected)) ) {
+                    $sectionclass .= ' current ' . $selected;
+                }
+                $text .= html_writer::start_tag('li', array('class' => $sectionclass));
+
+                $text .= html_writer::span('>', 'section-number');
+                if (!empty($globalconfig->display_course_link_text)) {
+                    $anchortext = $globalconfig->display_course_link_text;
+                } else {
+                    $anchortext = $course->shortname;
+                }
+
+                if ((!isset($selected)) && (empty($selected)) ) {
+                    $text .= '&nbsp; ' . $anchortext;
+                } else {
+                    $text .= '&nbsp; ' . html_writer::link(course_get_url($course), $anchortext);
+                }
+
+                $text .= html_writer::end_tag('li');
+            }
+
             $odd = $r % 2;
-            if ($format->is_section_current($section)) {
```

Comment:
This seems to remove the existing feature of highlighting the current section?

Reply:
From our testing it seems to highlight fine?

Reply:
I just re-checked and I believe there was some misunderstanding. Your has changed the meaning of the class "current" to match the section the user currently clicked to. But originally this was a section highlighted as the "current" one by the teacher of the course.

Anyway, I realized your intention and I agree that highlighting the currently selected section should be more prominent than highlighting the current one. I am adding an extra change that introduces a new CSS class "selected" which will use the original "current" styling. The "current" section will be still highlighted but not that much.

This is an example of the result of my fix: in the following example, the user is viewing the section "Topic 4" and the teacher has marked the "Topic 2" as the current one. Your patch did not highlight the "Topic 2" at all, my new version will show it in bold:

![image](https://user-images.githubusercontent.com/85831/35588101-ed16832a-05ff-11e8-930d-edef4a155f88.png)